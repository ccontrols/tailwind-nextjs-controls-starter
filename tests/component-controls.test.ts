import * as path from 'path';
import { loadConfigurations, extractDocuments, isMDXDocument } from '@component-controls/config';
import { renderExample, renderErr } from '@component-controls/test-renderers';
import { render, act } from '@testing-library/react';

describe('component-controls generated', () => {
  const configPath = path.resolve(__dirname, '../.storybook');
  const config = loadConfigurations(configPath);
  const documents = extractDocuments({ config, configPath });
  if (documents) {
    documents
      .filter((file: string) => !isMDXDocument(file, config.instrument))
      .forEach((file: string) => {
        // eslint-disable-next-line @typescript-eslint/no-var-requires, global-require, import/no-dynamic-require
        const exports = require(file);
        const doc = exports.default;
        const examples = Object.keys(exports)
          .filter((key) => key !== 'default')
          .map((key) => exports[key]);

        if (examples.length) {
          describe(doc.title, () => {
            examples.forEach((example) => {
              it(example.name, async () => {
                let rendered;
                act(() => {
                  rendered = renderExample({
                    example,
                    doc,
                    config,
                  });
                });
                if (!rendered) {
                  renderErr();
                  return;
                }
                const { asFragment } = render(rendered);
                expect(asFragment()).toMatchSnapshot();
              });
            });
          });
        }
      });
  }
});
