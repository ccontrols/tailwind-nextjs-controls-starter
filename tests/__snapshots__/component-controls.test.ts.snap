// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component-controls generated Components/Button bound Template 1`] = `
<DocumentFragment>
  <button
    class="rounded-md focus:outline-none focus:shadow-outline bg-primary-500 text-white hover:bg-primary-600 py-4 px-8"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;

exports[`component-controls generated Components/Button bound Template 2`] = `
<DocumentFragment>
  <button
    class="rounded-md focus:outline-none focus:shadow-outline bg-green-500 text-white hover:bg-green-600 py-4 px-8"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;

exports[`component-controls generated Components/Button bound Template 3`] = `
<DocumentFragment>
  <button
    class="rounded-md focus:outline-none focus:shadow-outline bg-yellow-500 text-white hover:bg-yellow-600 py-4 px-8"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;

exports[`component-controls generated Components/Button bound Template 4`] = `
<DocumentFragment>
  <button
    class="rounded-md focus:outline-none focus:shadow-outline bg-red-500 text-white hover:bg-red-600 py-4 px-8"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;

exports[`component-controls generated Components/Button bound Template 5`] = `
<DocumentFragment>
  <button
    class="rounded-md focus:outline-none focus:shadow-outline bg-primary-500 text-white hover:bg-primary-600 py-4 px-8 w-full"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;

exports[`component-controls generated Components/Button bound Template 6`] = `
<DocumentFragment>
  <button
    class="rounded-md focus:outline-none focus:shadow-outline bg-primary-500 text-white hover:bg-primary-600 py-4 px-8"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;

exports[`component-controls generated Components/Button bound Template 7`] = `
<DocumentFragment>
  <button
    class="rounded-md focus:outline-none focus:shadow-outline bg-primary-500 text-white hover:bg-primary-600 py-2 px-6"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;
